instrument.go

```diff
@@ -0,0 +1,49 @@
+package tally
+
+import (
+	"fmt"
+)
+
+// NewInstrumentedCall returns an InstrumentedCall with the given name
+func NewInstrumentedCall(scope Scope, name string) InstrumentedCall {
+	return &instrumentedCall{
+		error:   scope.Tagged(map[string]string{"result_type": "error"}).Counter(name),
+		success: scope.Tagged(map[string]string{"result_type": "success"}).Counter(name),
+		timing:  scope.Timer(fmt.Sprintf("%s.latency", name)),
+	}
+}
+
+var defaultSuccessFilter = func(err error) bool {
+	return err == nil
+}
+
+type instrumentedCall struct {
+	scope   Scope
+	success Counter
+	error   Counter
+	timing  Timer
+}
+
+// Exec executes the given block of code, and records whether it succeeded or
+// failed, and the amount of time that it took
+func (call *instrumentedCall) Exec(f ExecFn) error {
```

Comment:
nit: To standardize could you use a single letter for these receiver types, e.g. `func (c *instrumentedCall) ...`